metabase/metabase
# Disable saving models without up-to-date result_metadata

Fixes https://github.com/metabase/metabase/issues/37009

Disables creating new and updating existing models without `result_metadata`. That means that if the query was changed, the user would have to run the query first and wait for results.

How to verify:
- New -> Modal -> Native
- SELECT * FROM PRODUCTS
- Try to save the model without running the query. The button should be disabled and there should be a tooltip.
- Run the query
- Now the save button should become enabled
- Save the model
- ... menu -> Edit query definition
- Try changing the model query. The button should become disabled
- Run the updated query. The button should become enabled

<img width="1727" alt="Screenshot 2024-03-15 at 16 23 04" src="https://github.com/metabase/metabase/assets/8542534/3715f83d-482f-4da3-bfb7-547785dcf8fb">
